nifi-nar-bundles/nifi-ldap-iaa-providers-bundle/nifi-ldap-iaa-providers/src/main/java/org/apache/nifi/ldap/tenants/LdapUserGroupProvider.java

```diff
@@ -512,38 +523,66 @@ protected User doMapFromContext(DirContextOperations ctx) {
                                             + "misconfiguration or the user may just not belong to any groups. Ignoring group membership.", userGroupNameAttribute, identity));
                                 } else {
                                     try {
-                                        final NamingEnumeration<String> groupValues = (NamingEnumeration<String>) attributeGroups.getAll();
-                                        while (groupValues.hasMoreElements()) {
-                                            final String groupValue = groupValues.next();
-
-                                            // if we are performing a group search, then we need to normalize the group value so that each
-                                            // user associating with it can be matched. if we are not performing a group search then these
-                                            // values will be used to actually build the group itself. case sensitivity is for group
-                                            // membership, not group identification.
-                                            final String groupValueNormalized;
-                                            if (performGroupSearch) {
-                                                groupValueNormalized = groupMembershipEnforceCaseSensitivity ? groupValue : groupValue.toLowerCase();
-                                            } else {
-                                                groupValueNormalized = groupValue;
-                                            }
-
-                                            // store the group -> user identifier mapping... if case sensitivity is disabled, the group reference value will
-                                            // be lowercased when adding to groupToUserIdentifierMappings
-                                            groupToUserIdentifierMappings.computeIfAbsent(groupValueNormalized, g -> new HashSet<>()).add(user.getIdentifier());
-                                        }
+                                        NamingEnumeration<String> groupValuesFromAttribute = (NamingEnumeration<String>) attributeGroups.getAll();
+                                        groupValues = new LinkedList<String>();
+                                        while (groupValuesFromAttribute.hasMoreElements())
+                                            groupValues.add(groupValuesFromAttribute.next());
                                     } catch (NamingException e) {
                                         throw new AuthorizationAccessException("Error while retrieving user group name attribute [" + userIdentityAttribute + "].");
                                     }
                                 }
                             }
+                            if (StringUtils.isNotBlank(userGroupsFilterExpression)) {
```

Comment:
The property check on line 351 ensures that only one of the two group value determination paths is followed, but the logic here makes it appear that both could be followed.  Recommend changing this an "else if" for clarity.